### PR TITLE
Add GitHub Release Publish

### LINKED ISSUE #5857: Add GitHub Release for Copybook Tool
**Description:**
>$subject

Currently the GitHub release is not published in Copybook tool module. Need to update the gradle script.

### PR DESCRIPTION
**Pull request by DimuthuMadushan**
## Purpose
Fixes: [#5857](https://github.com/ballerina-platform/ballerina-library/issues/5857)

## Examples
N/A

## Checklist
- [X] Linked to an issue
- [ ] ~Updated the changelog~
- [ ] ~Added tests~
- [ ] ~Updated the spec~
- [ ] ~Checked native-image compatibility~

### CODE REVIEW

File: .github/workflows/dev-stage-central-publish.yml
Comment (ThisaruGuruge) [inline .github/workflows/dev-stage-central-publish.yml:1]: Can't we use the workflow templates for these?

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -25,6 +16,24 @@ jobs:
...
+          GITHUB_TOKEN: ${{ secrets.BALLERINA_BOT_TOKEN }}
+        run: |
+          echo "Version: ${VERSION}"
+          git checkout -b release-${VERSION}
+          sed -i 's/ballerinaLangVersion=\(.*\)-SNAPSHOT/ballerinaLangVersion=\1/g' gradle.properties
+          sed -i 's/ballerinaLangVersion=\(.*\)-[0-9]\{8\}-[0-9]\{6\}-.*$/ballerinaLangVersion=\1/g' gradle.properties
+          sed -i 's/stdlib\(.*\)=\(.*\)-SNAPSHOT/stdlib\1=\2/g' gradle.properties
+          sed -i 's/stdlib\(.*\)=\(.*\)-[0-9]\{8\}-[0-9]\{6\}-.*$/stdlib\1=\2/g' gradle.properties
+          sed -i 's/observe\(.*\)=\(.*\)-SNAPSHOT/observe\1=\2/g' gradle.properties
+          sed -i 's/observe\(.*\)=\(.*\)-[0-9]\{8\}-[0-9]\{6\}-.*$/observe\1=\2/g' gradle.properties
```
Comment (ThisaruGuruge) [inline .github/workflows/publish-release.yml]: Do we need these?

File: .github/workflows/dev-stage-central-publish.yml
Comment (DimuthuMadushan) [reply .github/workflows/dev-stage-central-publish.yml:1]: There is this additional step `Build the final Copybook tools package using Ballerina` which should complete with the gradle build. These gradle script related improvements are skipped for the initial implementation, and created the [issue](https://github.com/ballerina-platform/ballerina-library/issues/5776) to track. After resolving these, templates can be used.

File: .github/workflows/publish-release.yml (hunk 1 above)
Comment (DimuthuMadushan) [reply .github/workflows/publish-release.yml]: Can remove `stdlib` and `observe` changes.

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**